backend/src/main/java/com/bakdata/conquery/sql/conversion/cqelement/aggregation/DateAggregationTables.java

```diff
@@ -21,8 +21,11 @@ public String cteName(DateAggregationStep dateAggregationStep) {
 
 	public String getFromTableOf(DateAggregationStep dateAggregationStep) {
 		DateAggregationStep predecessor = dateAggregationStep.predecessor();
-		if (predecessor == null) {
-			return joinedTable;
+		while (!this.cteNameMap.containsKey(predecessor)) {
```

Comment:
Da sich das in den *Tables Klassen gleicht (ebenso wie andere Methoden) würde ich noch eine abstrakte Klasse als Basisklasse einführen.